### ISSUE TITLE
[Bug]: terminal not opening on 4.9.0 on macOS (github release and homebrew)

### ISSUE THREAD

**Issue by jsjoeio** [coder/code-server#5843]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Web Browser: Chrome
- Local OS: macOS
- Remote OS: macOS
- Remote Architecture: n/a
- `code-server --version`: 4.9.0

### Steps to Reproduce

1. `brew upgrade code-server`
2. `code-server`
3. open Integrated Terminal

### Expected

The terminal should load

### Actual

It doesn't load

### Logs

```typescript
  ERR TypeError: Cannot read properties of undefined (reading 'cell')
    at c.getFont (terminalConfigHelper.ts:188:73)
    at Ui.getFont (xtermTerminal.ts:416:29)
    at Ki._getDimension (terminalInstance.ts:698:40)
    at Ki._evaluateColsAndRows (terminalInstance.ts:649:26)
    at Ki.layout (terminalInstance.ts:1875:30)
    at Ki._open (terminalInstance.ts:1072:9)
    at Ki.setVisible (terminalInstance.ts:1350:9)
    at g.setInput (terminalEditor.ts:78:39)
    at async a.doSetInput (editorPanes.ts:353:4)
    at async a.doOpenEditor (editorPanes.ts:222:34)
```

### Screenshot/Video

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/206257187-ba3e23c5-3180-4939-b3ce-c1642d41f38f.png">

### Does this issue happen in VS Code or GitHub Codespaces?

- [X] I cannot reproduce this in VS Code.
- [X] I cannot reproduce this in GitHub Codespaces.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

**Comment by jsjoeio** [comment]
First thought is to test the macOS binary in the releases to see if this is specific to using code-server on Homebrew.

**Comment by jsjoeio** [comment]
Confirmed also broken on macOS release.

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/206259603-87c107f8-2b62-4d5b-bcf7-bc54e88478db.png">

**Comment by jsjoeio** [comment]
However, it is working on Linux so that's good. 

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/206259786-4b8aab3b-6092-42ba-9f28-5ce0fac2b61d.png">

I'll check npm next.

**Comment by jsjoeio** [comment]
npm is also broken, which means anyone on AUR or Homebrew will also have a broken terminal

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/206260328-a91d8b14-78c2-417b-8d22-131700e65c88.png">

**Comment by jsjoeio** [comment]
Since I develop code-server with code-server and Homebrew doesn't maintain versions, I'll have to download via npm and downgrade to 4.8.3

**Comment by jsjoeio** [comment]
Hmm...terminal works just fine in development 🤔 

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/206264528-a28d9cf8-c2eb-4ba3-a565-269f31853f71.png">

So something must be up with the build process.

**Comment by jsjoeio** [comment]
Hmm...so that one works:

1. download artifact
2. unzip
3. `cd release && yarn`
4.  `node . --port 3002`

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/206265712-ab57c5eb-cef2-4673-965d-21cf94078608.png">

To be clear this is the npm artifact after we modify the version 🤔 That could mean something during the macOS build and then also when we download the artifact before publishing to npm.

**Comment by jsjoeio** [comment]
Next step is to try replicating the "publish to npm" steps by downloading the `package.tar.gz` from releases and testing locally.

**Comment by jsjoeio** [comment]
Tried this: 

1. download package.tar.gz from 4.9.0 release
2. unzip
3. cd && yarn
4. node . --port 3002

and it worked as expected

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/206268756-132197ca-38f6-469b-bc48-1e353a61d1af.png">

Very confused 🤔

**Comment by jsjoeio** [comment]
Fixed via https://github.com/coder/code-server/pull/5844

**Comment by jsjoeio** [comment]
Still happening but i did notice that if i use `yarn` after downloading the github release for macOS, it works. We'll have to investigate further.

**Comment by jsjoeio** [comment]
## Investigation Notes

Part of me thinks I should revert [these changes](https://github.com/coder/code-server/pull/5844/files) and build a standalone release on my macOS and see if that works.

**Comment by jsjoeio** [comment]
##  Investigation - round 2 

## So when developing locally against `main` and running `yarn watch` it works 🤔 

Interesting...Want to try copying those `lib/vscode/node_modules` into the 4.9.1's directory and see if that works? 

## Sure.

**Comment by jsjoeio** [comment]
## Investigation Notes - round 3

So turns it out to be a discreprency with `xterm` versions, similar to this https://github.com/coder/code-server/issues/5530

We just tried another run and `ripgrep` failed in CI so none of `lib/vscode/node_modules` were installed but CI still passed and our native modules test passed so we need to change that back to `--list-extensions`.

Re-running again to see what happens
https://github.com/coder/code-server/actions/runs/3661582333

**Comment by jsjoeio** [comment]
TURNS OUT THAT WORKED.

<img width="1712" alt="image" src="https://user-images.githubusercontent.com/3806031/206814258-0ba89dee-7fdf-44ea-a239-73a53c948b55.png">

Followup for monday:
- [ ] add back `--list-extensions`  (--help only tests native modules for code-server, add note)
- [ ] run e2e on macOS runner (so we don't have to test manually)

**Comment by maxbane** [comment]
I just upgraded `code-server` to 4.9.0 on an Ubuntu 22.04.1 host (via `install.sh`), and the terminal no longer works in my iPadOS client. Thought I'd mention it here in case it's related (something about both clients using safari webkit?), but also happy to open a separate issue if preferable.

**Comment by jsjoeio** [comment]
Apologies! I think that's terminal gpu rendering related. I think I tagged you in another comment:

> Settings > Terminal > Gpu Acceleration "off" and now it renders correctly on iPad

Version has been fixed and will be out in the next release!

**Comment by jsjoeio** [comment]
Tested and verified this will be fixed in 4.9.1
